### PR TITLE
Account Service

### PR DESCRIPTION
**Pull request by KFloodBCGov**
# Description

This PR includes the following proposed change(s):

Account lookup service SOAP implementation added in.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

Local build and unit tests run (currently not passing)

Please describe the tests that you ran to verify your changes.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+        CsoAccountDetails csoAccountDetails = null;
+        if (StringUtils.isEmpty(userGuid)) return csoAccountDetails;
+
+        try {
+
+            AccountFacadeBean accountFacadeBeanPort = accountFacade.getAccountFacadeBeanPort();
+            List<ClientProfile> profiles = accountFacadeBeanPort.findProfiles(userGuid);
+
+            if (profiles.size() == 1) {
```
Comment (akroon3r) [inline src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java:34]: Why 1? What if its not 1?

File: src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                ClientProfile profile = profiles.get(0);
+                csoAccountDetails = new CsoAccountDetails(profile.getAccountId(), profile.getClientId());
+            }
+
+        }
+        catch(NestedEjbException_Exception e) {
+
+            LOGGER.error("Error calling findProfiles: " + e.getMessage());
```
Comment (akroon3r) [inline src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java:42]: Taylor was talking about this earlier, I think he said logging should be in the form of ```LOGGER.error('message', e)```

File: src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfiguration.java [hunk 3]
```diff
@@ -1,19 +1,51 @@
...
+        this.csoAccountProperties = csoAccountProperties;
+    }
+
+
+    public AccountFacade accountFacade() {
+
+        AccountFacade accountFacade = null;
+        try {
+
+            QName serviceName = new QName("http://accounts.csows.ag.gov.bc.ca/", "AccountFacade");
```
Comment (akroon3r) [inline src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfiguration.java:35]: dumb question: Are we ok with storing this URL in the repo

File: src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfiguration.java [hunk 4]
```diff
@@ -1,19 +1,51 @@
...
+            URL url = new URL(csoAccountProperties.getFilingAccountSoapUri());
+            accountFacade = new AccountFacade(url, serviceName);
+        } catch(MalformedURLException e) {
+
+            LOGGER.error("Malformed URL exception :" + e.getMessage());
```
Comment (akroon3r) [inline src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfiguration.java:40]: same as mentioned above

**Comment by akroon3r** [comment]
Congrats on PR #200

File: src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java (hunk 1 above)
Comment (KFloodBCGov) [reply src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java:34]: If we have more than one then we have no idea what we should do...  although I think there is potential for this.  Need to discuss with Jaime and figure out how to handle it.

Comment (akroon3r) [reply src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java:34]: Sounds good :)

File: src/backend/libs/efiling-account-client/src/test/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfigurationTest.java [hunk 5]
```diff
@@ -1,22 +1,32 @@
...
+
     @Test
     @DisplayName("CASE1: Test that beans are created")
     public void testBeansAreGenerated() {
-        context.run(it -> {
-            assertThat(it).hasSingleBean(EfilingAccountService.class);
+
+
+        CsoAccountProperties properties = new CsoAccountProperties();
+        properties.setFilingAccountSoapUri("http://localhost");
```
Comment (akroon3r) [inline src/backend/libs/efiling-account-client/src/test/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfigurationTest.java:24]: No port?

**Comment by akroon3r** [comment]
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/runner/work/jag-file-submission/jag-file-submission/src/backend/libs/efiling-account-client/src/test/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfigurationTest.java:[3,43] package com.sun.xml.internal.ws.wsdl.parser does not exist

File: src/backend/libs/efiling-account-client/src/test/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfigurationTest.java (hunk 5 above)
Comment (KFloodBCGov) [reply src/backend/libs/efiling-account-client/src/test/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfigurationTest.java:26]: It's a test that's meant to not be able to connect I think.

File: src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfiguration.java (hunk 3 above)
Comment (KFloodBCGov) [reply src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfiguration.java]: Yup, these are wsdl namespaces from the wsdl file itself (which is also checked in).

File: src/backend/libs/efiling-account-client/src/test/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfigurationTest.java (hunk 5 above)
Comment (akroon3r) [reply src/backend/libs/efiling-account-client/src/test/java/ca/bc/gov/open/jag/efilingaccountclient/config/AutoConfigurationTest.java]: Can't read apparently :|

File: src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java (hunk 2 above)
Comment (KFloodBCGov) [reply src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java]: Gotcha, and fixed.

**Pull request merged**